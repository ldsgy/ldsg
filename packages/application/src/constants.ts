import { ROOT_RESOURCE_KIND } from "@ldsg/constants";
import { HandlerSpecificResourceSettings } from "@ldsg/handler";
import { ResourceDefinitionSpecificResourceSettings } from "@ldsg/resource-definition";
import { GeneralResourceSettings } from "@ldsg/types";

/**
 * Instantiate Resource Handler General Resource Settings
 * 实例化资源处理程序普通资源配置
 */
export const INSTANTIATE_RESOURCE_HANDLER_GENERAL_RESOURCE_SETTINGS: GeneralResourceSettings =
  {
    title: "应用类型资源处理程序",
    description: "",
  };

/**
 * Instantiate Resource Handler Resource Settings
 * 实例化资源处理程序特殊资源配置
 * 框架通过此实例化此资源
 */
export const INSTANTIATE_RESOURCE_HANDLER_SPECIFIC_RESOURCE_SETTINGS: HandlerSpecificResourceSettings =
  {
    code: `export * from "@ldsg/application";`,
    dependencies: [
      {
        name: "@ldsg/application",
      },
    ],
  };

export const RESOURCE_DEFINITION_GENERAL_RESOURCE_SETTINGS: GeneralResourceSettings =
  { title: "应用类型资源定义", description: "" };

export const RESOURCE_DEFINITION_SPECIFIC_RESOURCE_SETTINGS: ResourceDefinitionSpecificResourceSettings =
  {
    kind: "application",
    parentKind: ROOT_RESOURCE_KIND,
  };
