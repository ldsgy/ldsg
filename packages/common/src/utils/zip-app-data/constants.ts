export const FILES_IN_HANDLER_MODELE = [
  {
    path: "src/index.ts",
  },
  {
    path: "package.json",
    data: {
      name: "PACKAGE-NAME",
      version: "1.0.0",
      main: "dist/index.js",
      scripts: {
        build: "tsc",
      },
      dependencies: {},
    },
  },
  {
    path: "tsconfig.json",
    data: {
      compilerOptions: {
        module: "commonjs",
        declaration: true,
        removeComments: true,
        emitDecoratorMetadata: true,
        experimentalDecorators: true,
        allowSyntheticDefaultImports: true,
        target: "ES2021",
        sourceMap: true,
        outDir: "./dist",
        baseUrl: "./",
        incremental: true,
        skipLibCheck: true,
        strictNullChecks: false,
        noImplicitAny: false,
        strictBindCallApply: false,
        forceConsistentCasingInFileNames: false,
        noFallthroughCasesInSwitch: false,
        resolveJsonModule: true,
        esModuleInterop: true,
        noEmitOnError: true,
      },
    },
  },
] as const;

export const FILES_IN_ROOT_MODELE = [
  {
    path: ".env",
  },
  {
    path: "index.ts",
    data: 'import { createApp } from "@ldsg/app";\nimport { SERVICE_TYPE_MAP } from "@ldsg/services";\nimport { StorageService } from "@ldsgy/storage-service";\nimport SERVICE_RECORDS_JSON from "./service-records.json";\n\nconst app = createApp({\n  serviceRecords: SERVICE_RECORDS_JSON,\n  serviceTypeMap: {\n    ...SERVICE_TYPE_MAP,\n    STORAGE: {\n      class: StorageService,\n    },\n  },\n});\n\nconst port = process.env.PORT || "3000";\n\napp.listen(port, () => {\n  console.log(`App listening on http://localhost:${port}`);\n\n  console.log(`GraphQL listening on http://localhost:${port}/graphql`);\n});\n',
  },
  {
    path: "manifest.json",
  },
  {
    path: "package.json",
    data: {
      name: "ldsg-app",
      version: "0.0.0",
      description: "",
      author: "",
      private: true,
      license: "UNLICENSED",
      scripts: {
        build:
          "pnpm run --workspace-root build:workspace-root && pnpm run --workspace-root build:recursive",
        "build:workspace-root": "tsc --project tsconfig.json",
        "build:recursive": "pnpm run --recursive build",
        start: "node dist/index.js",
      },
      dependencies: {
        "@ldsg/app": "latest",
        graphql: "^16.9.0",
        "graphql-middleware": "^6.1.35",
        "graphql-shield": "^7.6.5",
      },
      devDependencies: {
        "@types/node": "^22.1.0",
        "ts-loader": "^9.5.1",
        tslib: "^2.6.3",
        typescript: "^5.5.4",
      },
    },
  },
  {
    path: "pnpm-workspace.yaml",
    data: `packages:
  - "handlers/*"
`,
  },
  {
    path: "tsconfig.json",
    data: {
      compilerOptions: {
        module: "commonjs",
        declaration: true,
        removeComments: true,
        emitDecoratorMetadata: true,
        experimentalDecorators: true,
        allowSyntheticDefaultImports: true,
        target: "ES2021",
        sourceMap: true,
        outDir: "./dist",
        baseUrl: "./",
        incremental: true,
        skipLibCheck: true,
        strictNullChecks: false,
        noImplicitAny: false,
        strictBindCallApply: false,
        forceConsistentCasingInFileNames: false,
        noFallthroughCasesInSwitch: false,
        resolveJsonModule: true,
        esModuleInterop: true,
      },
      include: ["./index.ts", "./manifest.json"],
    },
  },
] as const;
