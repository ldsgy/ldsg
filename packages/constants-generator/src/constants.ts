import * as applicationModule from "@ldsg/application";
import * as handlerModule from "@ldsg/handler";
import * as resourceDefinitionModule from "@ldsg/resource-definition";
import { ResourceRecord } from "@ldsg/types";

export const ROOT_RESOURCE_ID = "root";

/**
 * 基座级资源模块
 */
export const BASE_RESOURCE_MODULES = [resourceDefinitionModule, handlerModule];

export const BASE_RESOURCE_KINDS = BASE_RESOURCE_MODULES.map(
  (baseResourceModule) =>
    baseResourceModule.RESOURCE_DEFINITION_SPECIFIC_RESOURCE_SETTINGS.kind
);

/**
 * 应用级资源模块
 */
export const APPLICATION_RESOURCE_MODULES = [applicationModule];

/**
 * 资源模块
 */
export const RESOURCE_MODULES = [
  ...BASE_RESOURCE_MODULES,
  ...APPLICATION_RESOURCE_MODULES,
];

export const MANIFEST_LIST: {
  name: string;
  resourceRecords: ResourceRecord[];
}[] = [
  {
    name: "app",
    resourceRecords: [
      {
        id: "main-app",
        kind: "application",
        parentId: ROOT_RESOURCE_ID,
        settings: {
          title: "主要应用",
          description: "",
        },
      },
    ],
  },
];
