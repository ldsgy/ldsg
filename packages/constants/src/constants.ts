import * as application from "@ldsg/application";
import * as handler from "@ldsg/handler";
import { Manifest } from "@ldsg/resource";
import * as resourceDefinition from "@ldsg/resource-definition";
import _ from "lodash";

export const ROOT_RESOURCE_ID = "root";

/**
 * 基础层资源类型组
 */
export const BASE_RESOURCE_KINDS = [resourceDefinition, handler];

/**
 * 应用层资源类型组
 */
export const APPLICATION_RESOURCE_KINDS = [application];

export const RESOURCE_KINDS = [
  ...BASE_RESOURCE_KINDS,
  ...APPLICATION_RESOURCE_KINDS,
];

export const RESOURCE_KINDS_RESOURCES: Manifest.Resource[] = _.flatMap(
  RESOURCE_KINDS.map((value) => {
    const { resourceDefinitionResourceSettings, handlerResourceSettings } =
      value;

    const resourceKind = _.kebabCase(resourceDefinitionResourceSettings.kind);

    const resourceDefinitionResourceId = `${resourceKind}-resource-definition`;

    const handlerResourceId = `${resourceKind}-resource-handler`;

    const res: Manifest.Resource[] = [
      {
        id: resourceDefinitionResourceId,
        kind: "RESOURCE_DEFINITION",
        parentId: ROOT_RESOURCE_ID,
        settings: resourceDefinitionResourceSettings,
      },
      {
        id: handlerResourceId,
        kind: "HANDLER",
        parentId: resourceDefinitionResourceId,
        settings: handlerResourceSettings,
      },
    ];

    return res;
  })
);
