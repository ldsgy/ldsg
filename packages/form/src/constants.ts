import { RESOURCE_DEFINITION_SPECIFIC_RESOURCE_SETTINGS as parentResourceDefinitionSpecificResourceSettings } from "@ldsg/forms";
import { HandlerSpecificResourceSettings } from "@ldsg/handler";
import { ResourceDefinitionSpecificResourceSettings } from "@ldsg/resource-definition";
import { GeneralResourceSettings } from "@ldsg/types";

export const HANDLER_GENERAL_RESOURCE_SETTINGS: GeneralResourceSettings = {
  title: "表单资源处理程序",
  description: "",
};

/**
 * Handler Resource Settings
 * 此类型资源的子级处理程序资源的配置
 * 如无或为 undefined，则说明无需子级处理程序。
 */
export const HANDLER_SPECIFIC_RESOURCE_SETTINGS: HandlerSpecificResourceSettings =
  {
    code: `export * from "@ldsg/form";`,
    dependencies: [
      {
        name: "@ldsg/form",
      },
    ],
  };

export const RESOURCE_DEFINITION_GENERAL_RESOURCE_SETTINGS: GeneralResourceSettings =
  { title: "表单资源定义", description: "" };

export const RESOURCE_DEFINITION_SPECIFIC_RESOURCE_SETTINGS: ResourceDefinitionSpecificResourceSettings =
  {
    kind: "form",
    parentKind: parentResourceDefinitionSpecificResourceSettings.kind,
  };
