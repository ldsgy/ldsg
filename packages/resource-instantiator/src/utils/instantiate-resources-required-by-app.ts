import { InstantiateResources, Resource } from "@ldsg/resource";
import { getResourceKindMap } from "./get-resource-kind-map";

/**
 * 实例化应用所需资源
 * 1. 实例化资源定义资源记录为资源实例
 * 2. 实例化非资源定义资源记录为资源实例
 */
export const instantiateResourcesRequiredByApp: InstantiateResources = (
  params
) => {
  const { resourceRecords } = params;

  const { resourceKindMap } = getResourceKindMap({
    resourceRecords,
  });

  const resources: Resource[] = resourceRecords.map((resourceRecord) => {
    const { kind } = resourceRecord;

    if (!(kind in resourceKindMap)) {
      throw new Error(`invalid kind '${kind}' in resource record`);
    }

    const { resourceDefinitionResource } = resourceKindMap[kind];

    const { resource } = resourceDefinitionResource.instantiateResource({
      resourceConstructorParams: resourceRecord,
    });

    return resource;
  });

  const res = {
    resources,
  };

  return res;
};
