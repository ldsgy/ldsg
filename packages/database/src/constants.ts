import { ROOT_RESOURCE_KIND } from "@ldsg/constants";
import { HandlerSpecificResourceSettings } from "@ldsg/handler";
import { ResourceDefinitionSpecificResourceSettings } from "@ldsg/resource-definition";
import { GeneralResourceSettings } from "@ldsg/types";

export const HANDLER_GENERAL_RESOURCE_SETTINGS: GeneralResourceSettings = {
  title: "数据库类型资源处理程序",
  description: "",
};

/**
 * Instantiate Resource Handler Resource Settings
 * 实例化资源处理程序特殊资源配置
 * 框架通过此实例化此资源
 */
export const INSTANTIATE_RESOURCE_HANDLER_SPECIFIC_RESOURCE_SETTINGS: HandlerSpecificResourceSettings =
  {
    code: `export * from "@ldsg/database";`,
    dependencies: [
      {
        name: "@ldsg/database",
      },
    ],
  };

export const RESOURCE_DEFINITION_GENERAL_RESOURCE_SETTINGS: GeneralResourceSettings =
  { title: "数据库类型资源定义", description: "" };

export const RESOURCE_DEFINITION_SPECIFIC_RESOURCE_SETTINGS: ResourceDefinitionSpecificResourceSettings =
  {
    kind: "database",
    parentKind: ROOT_RESOURCE_KIND,
  };
