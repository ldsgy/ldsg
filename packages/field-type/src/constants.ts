import { ROOT_RESOURCE_KIND } from "@ldsg/constants";
import { HandlerSpecificResourceSettings } from "@ldsg/handler";
import { ResourceDefinitionSpecificResourceSettings } from "@ldsg/resource-definition";
import { GeneralResourceSettings } from "@ldsg/types";
import { FieldTypeBasePlatform } from "./types/field-type-info";

/**
 * Instantiate Resource Handler General Resource Settings
 * 实例化资源处理程序普通资源配置
 */
export const INSTANTIATE_RESOURCE_HANDLER_GENERAL_RESOURCE_SETTINGS: GeneralResourceSettings =
  {
    title: "字段类型类型资源处理程序",
    description: "",
  };

/**
 * Instantiate Resource Handler Resource Settings
 * 实例化资源处理程序特殊资源配置
 * 框架通过此实例化此资源
 */
export const INSTANTIATE_RESOURCE_HANDLER_SPECIFIC_RESOURCE_SETTINGS: HandlerSpecificResourceSettings =
  {
    code: `export * from "@ldsg/field-type";`,
    dependencies: [
      {
        name: "@ldsg/field-type",
      },
    ],
  };

export const RESOURCE_DEFINITION_GENERAL_RESOURCE_SETTINGS: GeneralResourceSettings =
  { title: "字段类型类型资源定义", description: "" };

export const RESOURCE_DEFINITION_SPECIFIC_RESOURCE_SETTINGS: ResourceDefinitionSpecificResourceSettings =
  {
    kind: "field_type",
    parentKind: ROOT_RESOURCE_KIND,
  };

export const FieldTypeBasePlatformList = [
  FieldTypeBasePlatform.COMMON,
  FieldTypeBasePlatform.GRAPHQL,
  FieldTypeBasePlatform.JSON,
  FieldTypeBasePlatform.MONGOOSE,
];
