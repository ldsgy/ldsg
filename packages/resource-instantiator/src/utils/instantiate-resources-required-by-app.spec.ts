import { APP_MANIFEST } from "@ldsg/constants";
import { instantiateResourcesRequiredByApp } from "./instantiate-resources-required-by-app";

test("resolve-manifest-resources", () => {
  const { resourceRecords } = APP_MANIFEST;

  const { resources } = instantiateResourcesRequiredByApp({
    resourceRecords,
  });

  expect(resources).toMatchSnapshot();
});
