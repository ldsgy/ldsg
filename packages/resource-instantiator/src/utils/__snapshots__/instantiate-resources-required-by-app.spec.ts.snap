// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolve-manifest-resources 1`] = `
[
  ResourceDefinitionResource {
    "getFilteredResource": [Function],
    "getFilteredResources": [Function],
    "getHandler": [Function],
    "id": "resource-definition-resource-definition",
    "instantiateResource": [Function],
    "kind": "RESOURCE_DEFINITION",
    "parentId": "root",
    "settings": {
      "description": "",
      "handlerResourceId": "resource-definition-resource-handler",
      "kind": "RESOURCE_DEFINITION",
      "subKinds": [],
      "title": "资源定义类型资源定义",
    },
  },
  HandlerResource {
    "getFilteredResource": [Function],
    "getFilteredResources": [Function],
    "getHandler": [Function],
    "getModuleId": [Function],
    "id": "resource-definition-resource-handler",
    "kind": "HANDLER",
    "parentId": "root",
    "settings": {
      "code": "export * from "@ldsg/resource-definition";",
      "dependencies": [
        {
          "name": "@ldsg/resource-definition",
        },
      ],
      "description": "",
      "title": "资源定义类型资源处理程序",
    },
  },
  ResourceDefinitionResource {
    "getFilteredResource": [Function],
    "getFilteredResources": [Function],
    "getHandler": [Function],
    "id": "handler-resource-definition",
    "instantiateResource": [Function],
    "kind": "RESOURCE_DEFINITION",
    "parentId": "root",
    "settings": {
      "description": "主要包含引入模块列表与相应的处理程序代码，处理程序类型资源无子级资源。",
      "handlerResourceId": "handler-resource-handler",
      "kind": "HANDLER",
      "subKinds": [],
      "title": "处理程序类型资源定义",
    },
  },
  HandlerResource {
    "getFilteredResource": [Function],
    "getFilteredResources": [Function],
    "getHandler": [Function],
    "getModuleId": [Function],
    "id": "handler-resource-handler",
    "kind": "HANDLER",
    "parentId": "root",
    "settings": {
      "code": "export * from "@ldsg/handler";",
      "dependencies": [
        {
          "name": "@ldsg/handler",
        },
      ],
      "description": "包含处理程序相应的模块，此模块主要通过",
      "title": "处理程序类型资源处理程序",
    },
  },
  ResourceDefinitionResource {
    "getFilteredResource": [Function],
    "getFilteredResources": [Function],
    "getHandler": [Function],
    "id": "application-resource-definition",
    "instantiateResource": [Function],
    "kind": "RESOURCE_DEFINITION",
    "parentId": "root",
    "settings": {
      "description": "",
      "handlerResourceId": "application-resource-handler",
      "kind": "APPLICATION",
      "subKinds": [],
      "title": "应用类型资源定义",
    },
  },
  HandlerResource {
    "getFilteredResource": [Function],
    "getFilteredResources": [Function],
    "getHandler": [Function],
    "getModuleId": [Function],
    "id": "application-resource-handler",
    "kind": "HANDLER",
    "parentId": "root",
    "settings": {
      "code": "export * from "@ldsg/application";",
      "dependencies": [
        {
          "name": "@ldsg/application",
        },
      ],
      "description": "",
      "title": "应用类型资源处理程序",
    },
  },
  ApplicationResource {
    "createApp": [Function],
    "getFilteredResource": [Function],
    "getFilteredResources": [Function],
    "id": "main-app",
    "kind": "APPLICATION",
    "parentId": "root",
    "settings": {
      "description": "",
      "title": "主要应用",
    },
  },
]
`;
